### PR TITLE
Tracing create database in dn

### PR DESCRIPTION
**Pull request by xzxiong**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [x] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #https://github.com/matrixorigin/matrixone/issues/4931

## What this PR does / why we need it:
trace `create database` between CN and DN
we will have 1 statement_info record and 5 span_info records.

> PS: statement_info and span_info more detail in file https://github.com/matrixorigin/docs/blob/main/design/observability/how_to_use_trace_log_error.md

- 1 statement_info record
```
mysql> select * from statement_info where statement_id = '6b2fff04-6107-11ed-aa22-acde48001122'\G
*************************** 1. row ***************************
         statement_id: 6b2fff04-6107-11ed-aa22-acde48001122
       transaction_id: 66a81c0e-8cd6-4105-8a9d-5773b2055141
           session_id: 4b083098-6107-11ed-aa22-acde48001122
              account: abc
                 user: user1
                 host: 0.0.0.0
             database:
            statement: create database test
        statement_tag:
statement_fingerprint:
            node_uuid: 7c4dccb4-4d3c-41f8-b482-5251dc7a41bf
            node_type: ALL
           request_at: 2022-11-10 22:53:20.932554
          response_at: 2022-11-10 22:53:20.984573
             duration: 52018066
               status: Success
             err_code: 0
                error:
            exec_plan: {"code": 200, "message": "sql query no record execution plan", "steps": null, "success": false, "uuid": "6b2fff04-6107-11ed-aa22-acde48001122"}
            rows_read: 0
           bytes_scan: 0
1 row in set (0.10 sec)
```

- 5 span_info records
```
mysql> select trace_id, span_id, parent_span_id, span_name, start_time, end_time, duration from system.span_info where trace_id = '6b2fff04-6107-11ed-aa22-acde48001122';
+--------------------------------------+------------------+------------------+----------------------+----------------------------+----------------------------+----------+
| trace_id                             | span_id          | parent_span_id   | span_name            | start_time                 | end_time                   | duration |
+--------------------------------------+------------------+------------------+----------------------+----------------------------+----------------------------+----------+
| 6b2fff04-6107-11ed-aa22-acde48001122 | 808eef6497d8e4d0 | 0                | CreateDatabase       | 2022-11-10 22:53:20.956841 | 2022-11-10 22:53:20.957050 |   209655 |
| 6b2fff04-6107-11ed-aa22-acde48001122 | 613333ca461d8db3 | 12a184b4651624b3 | HandleCreateDatabase | 2022-11-10 22:53:20.957744 | 2022-11-10 22:53:20.957768 |    24163 |
| 6b2fff04-6107-11ed-aa22-acde48001122 | 12a184b4651624b3 | 833af34fa22ee025 | doWrite              | 2022-11-10 22:53:20.957475 | 2022-11-10 22:53:20.958132 |   656432 |
| 6b2fff04-6107-11ed-aa22-acde48001122 | 833af34fa22ee025 | 0                | doSend               | 2022-11-10 22:53:20.957284 | 2022-11-10 22:53:20.958237 |   953288 |
| 6b2fff04-6107-11ed-aa22-acde48001122 | 0ee4eefc9d7afb7f | 0                | doSend               | 2022-11-10 22:53:20.958257 | 2022-11-10 22:53:20.984514 | 26256108 |
+--------------------------------------+------------------+------------------+----------------------+----------------------------+----------------------------+----------+
5 rows in set (0.07 sec)
```

### CODE REVIEW

**Review by fengttt — APPROVED** [review]
This is great.

Do we know the cost of tracing?  Is there a tracing level like log verbosity level?

**Comment by xzxiong** [comment]
> This is great.
> 
> Do we know the cost of tracing? Is there a tracing level like log verbosity level?

cost of tracing
1. call `time.Now()` twice, cost 80ns each time. 
2. call `span.End()` will send span in queue. That may cost 1ms sometime in 15s cycle.

base on that cost, i should remove the tracing on `doWrite` and `doSend`...

**Comment by xzxiong** [comment]
> This is great.
> 
> Do we know the cost of tracing? Is there a tracing level like log verbosity level?

right now is only one interface `trace.Start(...)`.
but `trace.Debug(...)` seems necessary, let me find a way.

**Comment by xzxiong** [comment]
Add `trace.Debug(...)` , ctrl by config `enableTraceDebug`

**Comment by xzxiong** [comment]
@nnsgmsone you can see this usage as example

**Review by XuPeng-SH — APPROVED** [review]
lgtm

File: pkg/dnservice/store_rpc_handler.go [hunk 1]
```diff
@@ -22,6 +22,7 @@ import (
 	"github.com/matrixorigin/matrixone/pkg/pb/metadata"
```
Comment (zhangxu19830126) [inline pkg/dnservice/store_rpc_handler.go:22]: You can not add tracing here.  You should add trace in the txn/rpc/sender and server.go

File: pkg/txn/client/operator.go [hunk 2]
```diff
@@ -25,6 +25,7 @@ import (
 	"github.com/matrixorigin/matrixone/pkg/pb/txn"
```
Comment (zhangxu19830126) [inline pkg/txn/client/operator.go:25]: You can not add tracing here.  You should add trace in the txn/rpc/sender and server.go

**Review by zhangxu19830126 — CHANGES_REQUESTED** [review]

**Comment by xzxiong** [comment]
modification
1. move doSend to sender.doSender. `CN->DN`
2. move doRead to server.onMessage and Service.handleRPCRequest. `CN->LS`, `DN->LS`

**Review by zhangxu19830126 — APPROVED** [review]

**Review by aressu1985 — APPROVED** [review]

**Comment by xzxiong** [comment]
related issue #6728

**Pull request merged**